src/alfasim_sdk/_internal/alfacase/case_description.py

```diff
@@ -83,31 +83,89 @@ class PluginTableContainer:
     columns = attr.ib(default=attr.Factory(dict))
 
 
-@attr.s(frozen=True, slots=True)
-class TrendOutputDescription:
+@attr.s(kw_only=True)
+class _BaseTrendOutputDescription:
+    curve_names: List[str] = attr.ib(validator=list_of_strings)
+
+
+@attr.s()
+class PositionalTrendDescription(_BaseTrendOutputDescription):
     """
-    .. include:: /alfacase_definitions/TrendOutputDescription.txt
+    .. include:: /alfacase_definitions/PositionalTrendDescription.txt
 
     .. include:: /alfacase_definitions/list_of_unit_for_length.txt
     """
 
-    curve_names: List[str] = attr.ib(validator=list_of_strings)
     location = attrib_enum(type_=constants.OutputAttachmentLocation)
     position = attrib_scalar(default=None, is_optional=True, category="length")
     element_name: Optional[str] = attr.ib(
         default=None, validator=optional(instance_of(str))
     )
 
 
-@attr.s(frozen=True, slots=True)
+@attr.s()
+class GlobalTrendDescription(_BaseTrendOutputDescription):
+    """
+    .. include:: /alfacase_definitions/GlobalTrendDescription.txt
+    """
+
+    pass
+
+
+@attr.s()
+class OverallPipeTrendDescription(_BaseTrendOutputDescription):
+    """
+    .. include:: /alfacase_definitions/OverallPipeTrendDescription.txt
+    """
+
+    location = attrib_enum(type_=constants.OutputAttachmentLocation)
+    element_name: Optional[str] = attr.ib(
```

Comment:
I don't know why it was typed as Optional already, but all the trends must have the `element_name` defined (None is *not* valid for any type of trend that is tied to an element).

Reply:
Good point, I also don't know... `ProfileOutputDescription` also has an optional `element_name`, I'll remove the optional flag.